backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -896,10 +933,60 @@ public FutureExperimentRunDAO(
         executor);
   }
 
+  private InternalFuture<Map<String, VersioningEntry>> filterVersionedInputsBasedOnPrivileges(
+      Set<String> runIds, InternalFuture<Map<String, VersioningEntry>> futureVersionedInputs) {
+    return futureVersionedInputs.thenCompose(
+        versionedInputsMap -> {
+          List<InternalFuture<Map<String, VersioningEntry>>> internalFutureList = new ArrayList<>();
+          for (String runId : runIds) {
+            // Get VersionEntry from fetched map
+            VersioningEntry versioningEntry = versionedInputsMap.get(runId);
+            if (versioningEntry == null) {
+              continue;
+            }
+            String repoId = String.valueOf(versioningEntry.getRepositoryId());
+            // Check repository is accessible or not from versionedInputs If not then we will remove
+            // it from fetched map
+            internalFutureList.add(
+                InternalFuture.completedInternalFuture(versioningEntry)
```

Comment:
You don't have to create a future just to immediately use it. Just use the value directly.

Reply:
Fixed code